File: November 2025 GFG SOLUTION/November-28.java
class Solution {
    public static ArrayList<Integer> subsetXOR(int n) {
        ArrayList<Integer> ans = new ArrayList<>();
        int r = n & 3;
        int start = (r == 2) ? 2 : 1;
        int end = (r == 3) ? n - 1 : n;
        for (int i = start; i <= end; i++)
            if (r != 1 || i != n - 1) ans.add(i);
        return ans;
    }
}

File: November 2025 GFG SOLUTION/November-28.py
class Solution:
    def subsetXOR(self, n : int):
        ans, r = [], n & 3
        start = 2 if r == 2 else 1
        end = n - 1 if r == 3 else n
        for i in range(start, end + 1):
            if r != 1 or i != n - 1: ans.append(i)
        return ans
